src/Collection5.java

```diff
@@ -0,0 +1,55 @@
+import java.util.HashSet;
+
+import java.util.Set;
+public class Collection5 {
+
+    static class User {
+        private String name;
+        private int age;
+        public User(String name, int age) {
+            this.name = name;
+            this.age = age;
+        }
+        @Override
+        public int hashCode() {
+            char[] array = name.toCharArray();
+            int sum = 0;
+            for (int i =0; i< array.length-1;i++){
```

Comment:
in the loop it should be i < array.length to include the last character of the name in the hash calculation